File: 2486-MostFrequentEvenElement/2486-MostFrequentEvenElement.java
// Last updated: 7/30/2025, 12:22:45 PM
class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer,Integer> m = new HashMap<>();
        int value = 1000000;
        int frequency = 0;
        for (int i : nums){
            if(i%2==0){
                int count = m.getOrDefault(i,0) + 1;
                m.put(i,count);
                if( count>frequency || count==frequency && i<value){
                    value = i;
                    frequency = count ;
                }
            }
        }
        if(frequency == 0){
            return -1;
        }
        else {
            return value;
        }
    }
}

File: 2486-MostFrequentEvenElement/2486-MostFrequentEvenElement.py
# Last updated: 7/30/2025, 12:22:39 PM
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        nums = sorted(nums,reverse=True)
        ans = {}
        for i in nums:
            if i%2==0:
                if i in ans:
                    ans[i] += 1
                else :
                    ans[i] = 1
        m = 0
        res = -1
        for k,v in ans.items():
            if m <= v :
                m = v
                res = k
        return res

        